src/lib.rs

```diff
@@ -51,28 +54,56 @@ pub fn proxy(bin: &str) -> Result<()> {
 /// Minimalist argument parsing result type
 #[derive(PartialEq, Eq, Debug)]
 enum ArgsResult {
-    ExplicitSetup { use_local_bundle: Option<OsString> },
+    ExplicitSetup { use_local_bundle: Option<OsString>, use_local_toolchain: Option<OsString> },
     Default,
 }
 
 /// Parse `args` and decide what to do.
 fn parse_args(args: Vec<OsString>) -> ArgsResult {
-    // In an effort to keep our dependencies minimal, we do the bare minimum argument parsing manually.
-    // `args_ez` makes it easy to do crude arg parsing with match.
-    let args_ez: Vec<Option<&str>> = args.iter().map(|x| x.to_str()).collect();
-    // "cargo kani setup" comes in as "cargo-kani kani setup"
-    // "cargo-kani setup" comes in as "cargo-kani setup"
-    match &args_ez[..] {
-        &[_, Some("setup"), Some("--use-local-bundle"), _] => {
-            ArgsResult::ExplicitSetup { use_local_bundle: Some(args[3].clone()) }
-        }
-        &[_, Some("kani"), Some("setup"), Some("--use-local-bundle"), _] => {
-            ArgsResult::ExplicitSetup { use_local_bundle: Some(args[4].clone()) }
-        }
-        &[_, Some("setup")] | &[_, Some("kani"), Some("setup")] => {
-            ArgsResult::ExplicitSetup { use_local_bundle: None }
+    let matches = App::new("kani")
```

Comment:
Can you use the derive mode instead?

Reply:
Will add this in my cleanup issue and PR.